c7n/resources/elb_shared.py

```diff
@@ -0,0 +1,25 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+from c7n.utils import local_session
+from c7n.resources.vpc import SGPermission
+from c7n.filters import Filter
+
+class AdvancedIpPermissionsFilter(Filter):
+    # Can be overwritten by Child-Classes
+    permissions_key = None
+
+    def process(self, resources, event=None):
+        f = SGPermission(data = self.data, manager = self.manager)
+        f. ip_permissions_key = self.permissions_key
+        ec2 = local_session(self.manager.session_factory).client('ec2')
+        lbs = []
+        for r in resources:
+            if 'SecurityGroups' in r:
```

Comment:
I do wonder if we can be more efficient about this, either by leveraging the existing security group filter or directly fetching from the resource manager. the existing security group filters do seem like they would be useful since they contain the metadata for different resource types. ie. I think ingress/egress might be something we can expose generically across all network attached resources if they have an extant security group filter.

at the moment, if I have a 100 load balancers, with the same security group attached to them all, this will do a 100 api calls, even though one would have sufficed.

Reply:
put up a separate pr to cover the ingress/egress more generically #9382 and reuse extant security group filters.